orchestration/dap_orchestration/resources.py

```diff
@@ -0,0 +1,135 @@
+            "numWorkers": str(self.num_workers),
+            "maxNumWorkers": str(self.max_num_workers),
+            "project": self.google_project,
+            # hardcode anything that doesn't fall into solids/resources ie. dataflow
+            "runner": "dataflow",
+            # To use the service-based Dataflow Shuffle in your batch pipelines
+            "experiments": "shuffle_mode=service",
+        }
+
+    # pull the target_class out of the resource config
+
+    def run(
+            self,
+            run_arg_dict: dict[str, Any],
+    ) -> None:
+        # create a new dictionary containing the keys and values of arg_dict + solid arguments
+        dataflow_run_flags = {**self.arg_dict, **run_arg_dict}
+        self.logger.info("Dataflow beam runner")
+        # pull target_class out of list of flags
+        target_class = dataflow_run_flags.pop("target_class")
+        scala_project = dataflow_run_flags.pop("scala_project")
+
+        # list comprehension over args_dict to get flags
+        flags = " ".join([f'--{arg}={value}' for arg, value in dataflow_run_flags.items()])
+        subprocess.run(
+            # ["sbt", f'{self.scala_project}/runMain {self.target_class} {flags}'],
+            ["sbt", f'{scala_project}/runMain {target_class} {flags}'],
+            check=True,
+            cwd=self.working_dir
+        )
+
+
+@resource({
+    "working_dir": Field(StringSource),
+    "region": Field(StringSource),
+    "worker_machine_type": Field(StringSource),
+    "autoscaling_algorithm": Field(StringSource),
+    "num_workers": Field(IntSource),
+    "max_num_workers": Field(IntSource),
+    "google_project": Field(StringSource),
+})
+def dataflow_beam_runner(init_context: InitResourceContext) -> DataflowBeamRunner:
+    return DataflowBeamRunner(
+        working_dir=init_context.resource_config["working_dir"],
+        region=init_context.resource_config["region"],
+        worker_machine_type=init_context.resource_config["worker_machine_type"],
+        autoscaling_algorithm=init_context.resource_config["autoscaling_algorithm"],
+        num_workers=init_context.resource_config["num_workers"],
+        max_num_workers=init_context.resource_config["max_num_workers"],
+        google_project=init_context.resource_config["google_project"],
+        logger=init_context.log_manager,
+    )
+
+
+@dataclass
+class LocalBeamRunner:
+    working_dir: str
+    logger: DagsterLogManager
+
+    # todo: **kwargs param to local beam runner + dataflow beam runner when we refactor this to use dagster_utils
+
+    def __post_init__(self) -> None:
+        self.arg_dict = {
+            "runner": "direct",
+        }
+
+    # pull the target_class out of the resource config
+
+    def run(
+            self,
+            run_arg_dict: dict[str, Any],
+    ) -> None:
+        # create a new dictionary containing the keys and values of arg_dict + solid arguments
+        dataflow_run_flags = {**self.arg_dict, **run_arg_dict}
```

Comment:
🤔 This and the dataflow beam runner are almost the same in implementation, it appears they differ solely in the flags they consume. I'm wondering if we can extract a common class. Probably out of scope for this PR, but worth a small bite ticket so we don't forget.